### PR TITLE
Add customized dataset for cifar10 and cifar100

### PR DESCRIPTION
**Pull request by ZhaoPeiduo**

### CODE REVIEW

**Review by lizekai-richard — CHANGES_REQUESTED** [review]
This one is just an example given by the original repo. I believe we need to do some changes in our case. For example, the condition here should be the label encoding instead of the half image.

**Comment by ZhaoPeiduo** [comment]
> This one is just an example given by the original repo. I believe we need to do some changes in our case. For example, the condition here should be the label encoding instead of the half image.

Hmm do you have a list of things to change (or at least where I can look for things to customize)?

**Comment by lizekai-richard** [comment]
> > This one is just an example given by the original repo. I believe we need to do some changes in our case. For example, the condition here should be the label encoding instead of the half image.
> 
> Hmm do you have a list of things to change (or at least where I can look for things to customize)?

The most obvious one should be the condition. Also, the resolution might be different. We can discuss more in detail tomorrow.

File: datasets/cifar_custom.py [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+        image = pil_to_tensor(image)  / 255
+
+        # Standardize resolution to target size
+        if self.target_size:
+            resize_transform = transforms.Resize((self.target_size, self.target_size))
+            image = resize_transform(image)
+
+        if self.conditional:
+            # Not sure how to use labels as condition - specified by input_T?
+            condition = label
```
Comment (ZhaoPeiduo) [inline datasets/cifar_custom.py:57]: Use one hot encoding to represent the condition

File: datasets/cifar_custom.py [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+        if torch.is_tensor(idx):
+            idx = idx.tolist()
+
+        image, label = self.cifar[idx]
+        image = pil_to_tensor(image)  / 255
+
+        # Standardize resolution to target size
+        if self.target_size:
+            resize_transform = transforms.Resize((self.target_size, self.target_size))
+            image = resize_transform(image)
```
Comment (ZhaoPeiduo) [inline datasets/cifar_custom.py:53]: Delete this part since already handled by train_llm.py

**Pull request merged**